src/Entity/EntityDocument.php

```diff
@@ -31,4 +33,15 @@ public function getId();
 	 */
 	public function getType();
 
+	/**
+	 * Sets the id of the entity. If the id is not of the correct type,
+	 * an exception will be thrown. A specific derivative of EntityId is
+	 * always supported.
+	 *
+	 * @param EntityId|mixed $id
```

Comment:
Doing that will actually cause LSP violations. If you have an EntityId type that is not numeric, it can't construct itself from an integer, so this method won't be able to accept it.